### ISSUE TITLE
no such file to load -- require_relative

### ISSUE THREAD

**Issue by bkuhlmann** [mark-moseley/linecache#9]
While running a simple rake task in my Rails project (i.e. rake db:drop), I started getting the following stack dump:

rake aborted!
no such file to load -- require_relative
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/activesupport-3.0.8/lib/active_support/dependencies.rb:239:in `require'
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/activesupport-3.0.8/lib/active_support/dependencies.rb:239:in`require'
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/activesupport-3.0.8/lib/active_support/dependencies.rb:225:in `load_dependency'
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/activesupport-3.0.8/lib/active_support/dependencies.rb:596:in`new_constants_in'
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/activesupport-3.0.8/lib/active_support/dependencies.rb:225:in `load_dependency'
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/activesupport-3.0.8/lib/active_support/dependencies.rb:239:in`require'
/Users/brooke/.rvm/gems/ruby-1.8.7-p330/gems/linecache-0.45/lib/linecache.rb:66

Sure enough, the reference to "require_relative" did not make Rake happy. I ended up reverting back to linecache, version 0.43 in order to workaround this issue.

**Comment by annaswims** [comment]
I believe require_relative is only available in ruby 1.9.

**Comment by michaelklishin** [comment]
There is a [require_relative gem](https://rubygems.org/gems/require_relative) but [linecache.gemspec](https://github.com/mark-moseley/linecache/blob/master/linecache.gemspec) does not specify it as a dependency.

**Comment by michaelklishin** [comment]
Unfortunately, to properly fix this issue we also need to wait until [this require_relative issue](https://github.com/steveklabnik/require_relative/issues/5) is resolved.

**Comment by steveklabnik** [comment]
If you can't wait the day or two till I make a new release, you can always bundle your own edge version.

```
gem "require_relative", :git => "https://github.com/whatever/require_relative.git"
```

**Comment by michaelklishin** [comment]
I opened up a [pull request](https://github.com/mark-moseley/linecache/pull/11) with a fix

**Comment by jfirebaugh** [comment]
This was a bug in the [1.8.7 version of linecache](https://github.com/rocky/linecache), and was fixed in version 0.46. This repo is for the 1.9 version. So this issue can be closed.

**Comment by bkuhlmann** [comment]
:information_source: Closing this issue, per the above discussion, as it no longer needs to remain open.
